core/src/main/java/org/apache/accumulo/core/file/FileOperations.java

```diff
@@ -55,23 +56,24 @@ public static FileOperations getInstance() {
    */
 
   public abstract FileSKVIterator openReader(String file, Range range, Set<ByteSequence> columnFamilies, boolean inclusive, FileSystem fs, Configuration conf,
-      AccumuloConfiguration tableConf) throws IOException;
+      RateLimiter readLimiter, AccumuloConfiguration tableConf) throws IOException;
 
   public abstract FileSKVIterator openReader(String file, Range range, Set<ByteSequence> columnFamilies, boolean inclusive, FileSystem fs, Configuration conf,
-      AccumuloConfiguration tableConf, BlockCache dataCache, BlockCache indexCache) throws IOException;
+      RateLimiter readLimiter, AccumuloConfiguration tableConf, BlockCache dataCache, BlockCache indexCache) throws IOException;
 
   /**
    * Open a reader that fully support seeking and also enable any optimizations related to seeking, like bloom filters.
    *
    */
 
-  public abstract FileSKVIterator openReader(String file, boolean seekToBeginning, FileSystem fs, Configuration conf, AccumuloConfiguration acuconf)
-      throws IOException;
+  public abstract FileSKVIterator openReader(String file, boolean seekToBeginning, FileSystem fs, Configuration conf, RateLimiter readLimiter,
```

Comment:
> In many parts of our internal code, I would prefer we grab what we need from AccumuloConfiguration, combine it with any additional configuration appropriate for that specific context, and create a new context-specific configuration object to pass around within that context.

Somehow get the RateLimiterFactory directly from the config? That'd be a stop-gap.

> I was thinking perhaps a fluent syntax for opening readers/writers would make sense.
> That said, introducing such a change seemed a bit ambitious.

Yeah, I totally understand. Finding some middle ground to avoid lots of unrelated changes to your actual feature is ideal.